trail-register/src/main/java/com/rei/trailregister/FileUsageRepository.java

```diff
@@ -38,159 +39,163 @@
-		
-		int[] count = new int[] {0};
-		
-		withLock(compactionLock.readLock(), () -> {
-		    withLock(Paths.get(basedir.toString(), key.getApp(), key.getEnv(), key.getCategory(), key.getKey(), BASIC_ISO_DATE.format(date)), 
-		             Errors.rethrow().wrap(dateFile -> {
-		             
-        			if (!Files.exists(dateFile)) {
-        				Files.createDirectories(dateFile.getParent());
-        			} else {
-        				count[0] = readDataFile(dateFile);
-        			}
-        			Files.write(dateFile, String.valueOf(count[0]+num).getBytes());
-    		}));
-		});
-	}
+        checkArgument("app", key.getApp());
+        checkArgument("env", key.getEnv());
+        checkArgument("category", key.getCategory());
+        checkArgument("key", key.getKey());
+
+        int[] count = new int[] {0};
+
+        withLock(compactionLock.readLock(), () -> {
+            withLock(Paths.get(basedir.toString(), key.getApp(), key.getEnv(), key.getCategory(), key.getKey(), BASIC_ISO_DATE.format(date)),
+                     Errors.rethrow().wrap(dateFile -> {
+
+                    if (!Files.exists(dateFile)) {
+                        Files.createDirectories(dateFile.getParent());
+                    } else {
+                        count[0] = readDataFile(dateFile);
+                    }
+                    Files.write(dateFile, String.valueOf(count[0]+num).getBytes());
+            }));
+        });
+    }
 
     @Override
     public List<String> getApps() {
-		return list();
-	}
-	
-	@Override
+        return list();
+    }
+
+    @Override
     public List<String> getEnvironments(String app) {
-		return list(app);
-	}
-	
-	@Override
+        return list(app);
+    }
+
+    @Override
     public List<String> getCategories(String app, String env) {
-		return list(app, env);
-	}
-	
-	@Override
+        return list(app, env);
+    }
+
+    @Override
     public List<String> getKeys(String app, String env, String category) {
-		return list(app, env, category);
-	}
-	
-	private List<String> list(String... parts) {
-		try {
-			Path path = Paths.get(basedir.toString(), parts);
-			Preconditions.checkArgument(!path.toString().contains(".."), "may not contain '..' or '.'!");
-			
-			if (!Files.exists(path)) {
-				return Collections.emptyList();
-			}
-			return Files.list(path)
-					.filter(Files::isDirectory)
-					.map(Path::getFileName)
-					.map(Path::toString)
-					.collect(Collectors.toList());
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
-	
-	@Override
+        return list(app, env, category);
+    }
+
+    private List<String> list(String... parts) {
+        try {
+            Path path = Paths.get(basedir.toString(), parts);
+            Preconditions.checkArgument(!path.toString().contains(".."), "may not contain '..' or '.'!");
+
+            if (!Files.exists(path)) {
+                return Collections.emptyList();
+            }
+            Stream<Path> stream = Files.list(path);
```

Comment:
try-with-resources is the pattern that should be used here. It's more concise and makes sure `close()` gets called even if an IOException is thrown. That would look like this:

```
try (Stream<Path> stream = Files.list(path)) {
...
}
```

The close method gets automatically called because `Stream` implements `AutoCloseable`

Reply:
Looks like that does close the file! Updated